### PR TITLE
fix: dnd banner missing when active

### PR DESCRIPTION
**Pull request by sshanzel**
## Changes

Describe what this PR does
- refactored the DndContext to have a custom provider to have easier maintainability.
- moved from `usePersistentState` to `usePersistentContext`
- New property on main feed layout to accept custom banner.

Screen Shot 2022-03-02 at 2 35 34 PM

Screen Shot 2022-03-02 at 2 35 43 PM

Screen Shot 2022-03-02 at 2 35 50 PM

## Manual Testing

- On those affected packages:
- [x] Have you done sanity checks in the webapp?
- [x] Have you done sanity checks in the extension?

- Did you test the modified components media queries?
- [x] MobileL (420px)
- [x] Tablet (656px)
- [x] Laptop (1020px)

DD-563 #done

### CODE REVIEW

**Review by rebelchris — APPROVED** [review]
Looks great @sshanzel 

Just would like to ask you to test if the normal banners still work as intended and to double check the webapp with those.

File: packages/extension/src/newtab/DndContext.tsx [hunk 1]
```diff
@@ -0,0 +1,38 @@
+import usePersistentContext from '@dailydotdev/shared/src/hooks/usePersistentContext';
+import React, { ReactElement, ReactNode } from 'react';
+
+export type DndSettings = { expiration: Date; link: string };
+
+export interface DndContextData {
+  dndSettings: DndSettings;
+  isActive: boolean;
+  onDndSettings: (settings: DndSettings) => Promise<void>;
```
Comment (idoshamun) [inline packages/extension/src/newtab/DndContext.tsx:9]: Why did you decide to rename it to `on` instead of `set`?

**Review by idoshamun — APPROVED** [review]
Non blocking question. Other than that, we're good to go :)

File: packages/extension/src/newtab/DndContext.tsx (hunk 1 above)
Comment (sshanzel) [reply packages/extension/src/newtab/DndContext.tsx:9]: Just wanted to make things are uniform and easier to understand. We usually use the prefix `set` for state-related inside the component itself, to easily know that it is coming from either props or a destructured object.

**Pull request merged**